### ISSUE TITLE
libvirt-python: Error installing requirements-selftests

### ISSUE THREAD

**Issue by beraldoleal** [avocado-framework/avocado#3352]
If I try to install the `requirements-selftests` from the `Makefile`, it will try to install `libvirt-python==4.6.0`, and I will get the following error:

```
(avocado) [local@laptop avocado]$ make requirements-selftests
/home/local/src/venvs/avocado/bin/python3 -m pip --version || /home/local/src/venvs/avocado/bin/python3 -m ensurepip  || /home/local/src/venvs/avocado/bin/python3 -c "import os; import sys; import urllib; f = urllib.urlretrieve('https://bootstrap.pypa.io/get-pip.py')[0]; os.system('%s %s' % (sys.executable, f))"
pip 19.3.1 from /home/local/src/venvs/avocado/lib64/python3.7/site-packages/pip (python 3.7)
/home/local/src/venvs/avocado/bin/python3 -m pip install -r requirements-selftests.txt
Requirement already satisfied: pylint==2.3.0 in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from -r requirements-selftests.txt (line 4)) (2.3.0)
Requirement already satisfied: astroid==2.2.0 in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from -r requirements-selftests.txt (line 5)) (2.2.0)
Requirement already satisfied: inspektor==0.5.2 in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from -r requirements-selftests.txt (line 6)) (0.5.2)
Requirement already satisfied: funcsigs>=0.4 in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from -r requirements-selftests.txt (line 9)) (1.0.2)
Requirement already satisfied: aexpect==1.5.1 in /home/local/src/venvs/avocado/lib/python3.7/site-packages/aexpect-1.5.1-py3.7.egg (from -r requirements-selftests.txt (line 12)) (1.5.1)
Requirement already satisfied: psutil==5.4.7 in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from -r requirements-selftests.txt (line 13)) (5.4.7)
Requirement already satisfied: pycdlib==1.6.0 in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from -r requirements-selftests.txt (line 17)) (1.6.0)
Collecting libvirt-python==4.6.0
  Using cached https://files.pythonhosted.org/packages/28/d3/a04735d385ac645a758c89d1d1f4cbf72ae01ae4f85bc55d1de5af6254bb/libvirt-python-4.6.0.tar.gz
Requirement already satisfied: netifaces in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from -r requirements-selftests.txt (line 26)) (0.10.9)
Requirement already satisfied: elementpath==1.1.8 in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from -r requirements-selftests.txt (line 29)) (1.1.8)
Requirement already satisfied: xmlschema==1.0.11 in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from -r requirements-selftests.txt (line 30)) (1.0.11)
Requirement already satisfied: mccabe in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from pylint==2.3.0->-r requirements-selftests.txt (line 4)) (0.6.1)
Requirement already satisfied: isort>=4.2.5 in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from pylint==2.3.0->-r requirements-selftests.txt (line 4)) (4.3.21)
Requirement already satisfied: typed-ast>=1.3.0; python_version >= "3.7" and implementation_name == "cpython" in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from astroid==2.2.0->-r requirements-selftests.txt (line 5)) (1.4.0)
Requirement already satisfied: six in /home/local/src/venvs/avocado/lib/python3.7/site-packages/six-1.12.0-py3.7.egg (from astroid==2.2.0->-r requirements-selftests.txt (line 5)) (1.12.0)
Requirement already satisfied: wrapt in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from astroid==2.2.0->-r requirements-selftests.txt (line 5)) (1.11.2)
Requirement already satisfied: lazy-object-proxy in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from astroid==2.2.0->-r requirements-selftests.txt (line 5)) (1.4.3)
Requirement already satisfied: pycodestyle in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from inspektor==0.5.2->-r requirements-selftests.txt (line 6)) (2.5.0)
Requirement already satisfied: cliff in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from inspektor==0.5.2->-r requirements-selftests.txt (line 6)) (2.16.0)
Requirement already satisfied: cmd2!=0.8.3,<0.9.0,>=0.8.0 in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from cliff->inspektor==0.5.2->-r requirements-selftests.txt (line 6)) (0.8.9)
Requirement already satisfied: PrettyTable<0.8,>=0.7.2 in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from cliff->inspektor==0.5.2->-r requirements-selftests.txt (line 6)) (0.7.2)
Requirement already satisfied: pbr!=2.1.0,>=2.0.0 in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from cliff->inspektor==0.5.2->-r requirements-selftests.txt (line 6)) (5.4.3)
Requirement already satisfied: stevedore>=1.20.0 in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from cliff->inspektor==0.5.2->-r requirements-selftests.txt (line 6)) (1.31.0)
Requirement already satisfied: PyYAML>=3.12 in /home/local/src/venvs/avocado/lib/python3.7/site-packages/PyYAML-5.1.2-py3.7-linux-x86_64.egg (from cliff->inspektor==0.5.2->-r requirements-selftests.txt (line 6)) (5.1.2)
Requirement already satisfied: pyparsing>=2.1.0 in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from cliff->inspektor==0.5.2->-r requirements-selftests.txt (line 6)) (2.4.2)
Requirement already satisfied: pyperclip in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from cmd2!=0.8.3,<0.9.0,>=0.8.0->cliff->inspektor==0.5.2->-r requirements-selftests.txt (line 6)) (1.7.0)
Requirement already satisfied: wcwidth; sys_platform != "win32" in /home/local/src/venvs/avocado/lib/python3.7/site-packages (from cmd2!=0.8.3,<0.9.0,>=0.8.0->cliff->inspektor==0.5.2->-r requirements-selftests.txt (line 6)) (0.1.7)
Installing collected packages: libvirt-python
    Running setup.py install for libvirt-python ... error
    ERROR: Command errored out with exit status 1:
     command: /home/local/src/venvs/avocado/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-m77zz4zy/libvirt-python/setup.py'"'"'; __file__='"'"'/tmp/pip-install-m77zz4zy/libvirt-python/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-0_2wp0sd/install-record.txt --single-version-externally-managed --compile --install-headers /home/local/src/venvs/avocado/include/site/python3.7/libvirt-python
         cwd: /tmp/pip-install-m77zz4zy/libvirt-python/
    Complete output (11 lines):
    running install
    running build
    /usr/bin/pkg-config --print-errors --atleast-version=0.9.11 libvirt
    /home/local/src/venvs/avocado/bin/python3 generator.py libvirt /usr/share/libvirt/api/libvirt-api.xml
    Found 447 functions in /usr/share/libvirt/api/libvirt-api.xml
    Found 0 functions in libvirt-override-api.xml
    Generated 365 wrapper functions
    Missing type converters:
    virTypedParameterPtr:1
    ERROR: failed virDomainSetIOThreadParams
    error: command '/home/local/src/venvs/avocado/bin/python3' failed with exit status 1
    ----------------------------------------
ERROR: Command errored out with exit status 1: /home/local/src/venvs/avocado/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-m77zz4zy/libvirt-python/setup.py'"'"'; __file__='"'"'/tmp/pip-install-m77zz4zy/libvirt-python/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-0_2wp0sd/install-record.txt --single-version-externally-managed --compile --install-headers /home/local/src/venvs/avocado/include/site/python3.7/libvirt-python Check the logs for full command output.
make: [Makefile:129: requirements-selftests] Error 1 (ignored)

```

But if I install the latest libvirt-python (5.7.0) and comment the libvirt-python line from `requirements-selftests.txt`, everything goes fine.

Looking into the requirements-selftests.txt file, you can find a comment:

```
# this is a workaround since easy_install cannot install
# libvirt-python properly with the the current process
# in make develop. The proper solution would be migrate
# our make develop process to use pip
libvirt-python==4.6.0
```

Should we upgrade the version to 5.7.0?

**Comment by clebergnu** [comment]
When you say that you "install the latest libvirt-python (5.7.0)" what kind of installation are you performing?  If 5.7.0 installs cleanly via pip then it's probably an improvement.  We should also check if the installation will be done without issues on Travis-CI too.

**Comment by beraldoleal** [comment]
@clebergnu I just installed with the command:

```pip install libvirt-python```

**Comment by clebergnu** [comment]
I'm guessing this was fixed in #3366, right?  If not, please reopen.
